pandas/tools/tile.py

```diff
@@ -187,15 +189,27 @@ def qcut(x, q, labels=None, retbins=False, precision=3):
     bins = algos.quantile(x, quantiles)
     fac, bins = _bins_to_cuts(x, bins, labels=labels,
                               precision=precision, include_lowest=True,
-                              dtype=dtype)
+                              dtype=dtype, duplicates=duplicates)
 
     return _postprocess_for_cut(fac, bins, retbins, x_is_series,
                                 series_index, name)
 
 
 def _bins_to_cuts(x, bins, right=True, labels=None,
                   precision=3, include_lowest=False,
-                  dtype=None):
+                  dtype=None, duplicates='raise'):
+
+    if duplicates not in ['raise', 'drop']:
+        raise ValueError("invalid value for 'duplicates' parameter, "
+                         + "valid options are: raise, drop")
+
+    if len(algos.unique(bins)) < len(bins):
```

Comment:
Not sure I understand -- if I do that before line 213, how would I compare len of bins before and after in line 207 and how would I output the original non unique edges in 209 in case duplicates='raise'?